components/clarity-repl/src/repl/session.rs

```diff
@@ -551,6 +551,15 @@ impl Session {
         test_name: Option<String>,
         ast: &mut Option<ContractAST>,
     ) -> Result<ExecutionResult, Vec<Diagnostic>> {
+        if contract.epoch == StacksEpochId::Epoch20 && contract.clarity_version == ClarityVersion::Clarity2 {
```

Comment:
Good catch!
We have the same logic to handle clarity version and epoch that are set in the Clarinet.toml.
https://github.com/hirosystems/clarinet/blob/30e6807997be3071cfde6dff1f1851724eadd847/components/clarinet-files/src/project_manifest.rs#L163-L167

Could you add a similar condition?
It should help handling future epochs and clarity versions